crypto/src/main/java/org/web3j/crypto/RawTransaction.java

```diff
@@ -118,6 +120,64 @@ public static RawTransaction createTransaction(
                         maxFeePerGas));
     }
 
+    public static RawTransaction createTransaction(
+            long chainId,
+            BigInteger nonce,
+            BigInteger maxPriorityFeePerGas,
+            BigInteger maxFeePerGas,
+            BigInteger gasLimit,
+            String to,
+            BigInteger value,
+            String data,
+            BigInteger maxFeePerBlobGas,
+            List<Bytes> versionedHashes) {
+
+        return new RawTransaction(
+                Transaction4844.createTransaction(
+                        chainId,
+                        nonce,
+                        maxPriorityFeePerGas,
+                        maxFeePerGas,
+                        gasLimit,
+                        to,
+                        value,
+                        data,
+                        maxFeePerBlobGas,
+                        versionedHashes));
+    }
+
+    public static RawTransaction createTransaction(
+            List<Blob> blobs,
+            List<Bytes> kzgCommitments,
+            List<Bytes> kzgProofs,
+            long chainId,
+            BigInteger nonce,
+            BigInteger maxPriorityFeePerGas,
+            BigInteger maxFeePerGas,
+            BigInteger gasLimit,
+            String to,
+            BigInteger value,
+            String data,
+            BigInteger maxFeePerBlobGas,
+            List<Bytes> versionedHashes) {
```

Comment:
If I provide the blobs, couldn't it calculate some of these other parameters for me, like versionedHashes and proofs? Or am I getting this wrong? See my comment below on Transaction4844.java as well.

Reply:
Yes makes sense, but we are using this when decoding 4844 tx, that time we are decoding blobs, commitments, proofs and hashes. So we can directly decode and create a 4844 tx.